### PR TITLE
Move everything from Hud::Base into Structre::Base concern

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/concerns/hmis/structure/base.rb [hunk 1]
```diff
@@ -7,7 +7,107 @@
...
+      bipartite_keys col, model_name
+    end
+
+    def hud_key=(key)
+      @hud_key = key
+    end
+
+    def hud_key
+      @hud_key
+    end
```
Comment (awisema) [inline app/models/concerns/hmis/structure/base.rb:82]: `attr_accessor`?

File: app/models/concerns/hmis/structure/base.rb [hunk 2]
```diff
@@ -7,7 +7,107 @@
...
+
+    # Override in sub-classes as necessary
+    def hud_primary_key
+      hud_csv_headers.first
+    end
+
+    ## convenience methods to DRY up some association definitions
+
+    def bipartite_keys(col, model_name)
```
Comment (awisema) [inline app/models/concerns/hmis/structure/base.rb]: `model_name = nil`?

File: app/models/grda_warehouse/hud/site.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@
 module GrdaWarehouse::Hud
   class Site < Base
     include HudSharedScopes
+    include ::HMIS::Structure::Base
     acts_as_paranoid column: :DateDeleted
```
Comment (awisema) [inline app/models/grda_warehouse/hud/site.rb:11]: Should this use `hud_paranoid_column`?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/concerns/hmis/structure/base.rb (hunk 1 above)
Comment (eanders) [reply app/models/concerns/hmis/structure/base.rb:82]: class method, so nope

File: app/models/grda_warehouse/hud/site.rb (hunk 3 above)
Comment (eanders) [reply app/models/grda_warehouse/hud/site.rb:11]: It could.  This class is deprecated, so the fewer dependencies the better.

**Pull request merged**